cli/src/main/java/com/devonfw/tools/ide/tool/ToolCommandlet.java

```diff
@@ -291,202 +221,6 @@ protected void postInstall() {
-    } catch (IOException e) {
-      throw new IllegalStateException("Failed to write version file " + toolVersionFile, e);
-    }
-    return createToolInstallation(toolPath, resolvedVersion, toolVersionFile);
-  }
-
-  private ToolInstallation createToolInstallation(Path rootDir, VersionIdentifier resolvedVersion,
-      Path toolVersionFile) {
-
-    Path linkDir = getMacOsHelper().findLinkDir(rootDir);
-    Path binDir = linkDir;
-    Path binFolder = binDir.resolve(IdeContext.FOLDER_BIN);
-    if (Files.isDirectory(binFolder)) {
-      binDir = binFolder;
-    }
-    if (linkDir != rootDir) {
-      assert (!linkDir.equals(rootDir));
-      this.context.getFileAccess().copy(toolVersionFile, linkDir, FileCopyMode.COPY_FILE_OVERRIDE);
-    }
-    return new ToolInstallation(rootDir, linkDir, binDir, resolvedVersion);
-  }
-
-  private MacOsHelper getMacOsHelper() {
-
-    if (this.macOsHelper == null) {
-      this.macOsHelper = new MacOsHelper(this.context);
-    }
-    return this.macOsHelper;
-  }
-
-  /**
-   * @return {@code true} to extract (unpack) the downloaded binary file, {@code false} otherwise.
-   */
-  protected boolean isExtract() {
-
-    return true;
-  }
-
-  /**
-   * @return the currently installed tool version or {@code null} if not found (tool not installed).
-   */
-  protected String getInstalledToolVersion() {
-
-    Path toolPath = getToolPath();
-    if (!Files.isDirectory(toolPath)) {
-      this.context.debug("Tool {} not installed in {}", getName(), toolPath);
-      return null;
-    }
-    Path toolVersionFile = toolPath.resolve(IdeContext.FILE_SOFTWARE_VERSION);
-    if (!Files.exists(toolVersionFile)) {
-      Path legacyToolVersionFile = toolPath.resolve(IdeContext.FILE_LEGACY_SOFTWARE_VERSION);
-      if (Files.exists(legacyToolVersionFile)) {
-        toolVersionFile = legacyToolVersionFile;
-      } else {
-        this.context.warning("Tool {} is missing version file in {}", getName(), toolVersionFile);
-        return null;
-      }
-    }
-    try {
-      return Files.readString(toolVersionFile).trim();
-    } catch (IOException e) {
-      throw new IllegalStateException("Failed to read file " + toolVersionFile, e);
-    }
-  }
-
-  private boolean isInstalledVersion(VersionIdentifier expectedVersion, VersionIdentifier installedVersion,
-      boolean silent) {
-
-    if (expectedVersion.equals(installedVersion)) {
-      IdeLogLevel level = IdeLogLevel.INFO;
-      if (silent) {
-        level = IdeLogLevel.DEBUG;
-      }
-      this.context.level(level).log("Version {} of tool {} is already installed", installedVersion,
-          getToolWithEdition());
-      return true;
-    }
-    return false;
-  }
-
-  /**
-   * @param file the {@link Path} to the file to extract.
-   * @param targetDir the {@link Path} to the directory where to extract (or copy) the file.
-   */
-  protected void extract(Path file, Path targetDir) {
```

Comment:
I would expect the `extract` logic to be available both for local and global tool commandlets.
While most global tools directly download an MSI, PKG, EXE, dpkg, etc. and install it without that we extract it in our Java code, there can be some edge-cases, where we will also need this for global tool commandlets in the future...

Reply:
BTW: the `isExtract()` method is there from the old design of devonfw-ide in bash and represents the parameter
`noUnpack` that was mainly used for global tool commandlets:
https://github.com/devonfw/ide/blob/af67497e06fb82e2e50a4d9ffa10e3f9dda44aa1/scripts/src/main/resources/scripts/functions#L1266

Reply:
Done, local are by default extractable and global arent but in need we can override isExtract and make it return true

Reply:
Question now is in case we need to extract a global tool where should we extract it? In the download cache or somewhere else?

Reply:
> Question now is in case we need to extract a global tool where should we extract it? In the download cache or somewhere else?

This shall be done in a temp folder as it will only be needed temporary to run the installer and after the installation (e.g. to `C:\Program Files`) it is not needed anymore. This is the same behavior as in devonfw-ide.